### PR TITLE
Remove moq

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Moq added a closed source obfuscated DLL which wrote some sponsorship messages when building which has some privacy implications, https://github.com/moq/moq/issues/1372. This was later removed but this is a good time to remove a barely used dependency since we mostly use NSubstitute.

Do note that I'm not an expert in either Moq or NSubstitute. Everything seems to run as expected (except EF tests but that's due to DB schema changes).

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: test/Core.Test/Core.Test.csproj [hunk 1]
```diff
@@ -9,7 +9,6 @@
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="$(MicrosoftNetTestSdkVersion)" />
-    <PackageReference Include="Moq" Version="4.17.2" />
```
Comment (withinfocus) [inline test/Core.Test/Core.Test.csproj:12]: 🎨 Can you get the latest and also remove @justindbaur 's Renovate blocker too?

File: bitwarden_license/test/Commercial.Core.Test/packages.lock.json [hunk 2]
```diff
@@ -2675,75 +2666,74 @@
       "commercial.core": {
         "type": "Project",
         "dependencies": {
-          "Core": "2023.7.2"
+          "Core": "[2023.7.2, )"
```
[the remark this answers is not in the document]
Comment (Hinton) [reply bitwarden_license/test/Commercial.Core.Test/packages.lock.json:2669]: No idea honestly. This always happens when I refresh my packages in either Rider or Visual Studio.

**Review by withinfocus — APPROVED** [review]
Tests pass -- :shipit:

File: bitwarden_license/test/Commercial.Core.Test/packages.lock.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (MGibson1) [reply bitwarden_license/test/Commercial.Core.Test/packages.lock.json:2669]: I'm frustrated by this, too. I haven't been able to find a `pinned` flag or anything in `dotnet restore`

**Review by MGibson1 — APPROVED** [review]

**Pull request merged**